front/src/Utils/PathfindingManager.ts

```diff
@@ -122,12 +122,15 @@ export class PathfindingManager {
         this.easyStar.setAcceptableTiles([0]); // zeroes are walkable
     }
 
-    private setWalkingCostGrid(grid: number[][]): void {
-        for (let y = 0; y < grid.length; y += 1) {
-            for (let x = 0; x < grid[y].length; x += 1) {
-                this.easyStar.setAdditionalPointCost(x, y, grid[y][x]);
+    private async setWalkingCostGrid(grid: number[][]): Promise<void> {
+        return new Promise<void>((resolve, reject) => {
+            for (let y = 0; y < grid.length; y += 1) {
+                for (let x = 0; x < grid[y].length; x += 1) {
+                    this.easyStar.setAdditionalPointCost(x, y, grid[y][x]);
+                }
```

Comment:
Woah.
I was under the impression that the callback is called as soon as the constructor of "Promise" is called.
Are you sure the call to the callback is delayed?